### ISSUE TITLE
Support for FiniteDuration

### ISSUE THREAD

**Issue by etspaceman** [fthomas/refined#621]
#397 had some discussion about whether or not this library supports FiniteDuration. It seems to be a pretty standard use-case - I'd like to see some official support for it in this library.

**Comment by henricook** [comment]
I've just spent some time looking into how to write my own predicates for this, but looks hard!

**Comment by fthomas** [comment]
I wrote a bit about supporting `FiniteDuration` [here](https://github.com/fthomas/refined/issues/633#issuecomment-488071074). @etspaceman It would help if you could provide one or two examples how you would refine `FiniteDuration`.

**Comment by etspaceman** [comment]
Sorry for the late response here! I would largely treat it like a numeric instance myself. Things like `Positive`, `NonNegative`, etc would be my most highly used thing.

**Comment by fthomas** [comment]
Ok, this would be possible with the `LengthInSeconds` predicate mentioned in #633, i.e. `FiniteDuration Refined LengthInSeconds[NonNegative]`, etc.

**Comment by etspaceman** [comment]
Yes I've done similar things, but have dropped the usage as the `auto` package does not support FiniteDuration. That's mainly why I opened this ticket. Without `auto` support, using this for FiniteDuration literals becomes a big pain.
